source/arm7/libc/crt0_device_list.c

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: Zlib
+//
+// Copyright (C) 2025 Edoardo Lolletti (edo9300)
+
+#include <stddef.h>
+#include <string.h>
+
+#include <nds/system.h>
+#include <nds/device_list.h>
+
+static size_t getDeviceNameLenFromAppName(const char (appname)[0x40])
+{
+	for(int i = 0; i < 0x40; ++i)
+	{
+		if(appname[i] == ':')
+		{
+			return i;
+		}
+	}
+	return 0;
+}
+
+struct device_list_argv {
+	struct __argv base;
+    char argv[0x42];
+};
+
+void check_device_list()
+{
+	if (!isDSiMode())
+		return;
+
+	if (__system_argv->argvMagic == ARGV_MAGIC)
+		return;
+
+	DeviceList* deviceList = __DeviceList;
+	if (!deviceList)
+		return;
+
+	size_t deviceNameLen = getDeviceNameLenFromAppName(deviceList->appname);
+
+	if(deviceNameLen == 0)
+	{
+		return;
+	}
+
+	for (int i = 0; i < 11; ++i)
+	{
+		Device* device = &deviceList->devices[i];
+		if(device->deviceName[deviceNameLen] == '\0' && strncmp(device->deviceName, deviceList->appname, deviceNameLen) == 0)
+		{
+			struct device_list_argv* argv = (struct device_list_argv*)__system_argv;
+			argv->base.argvMagic = ARGV_MAGIC;
+			argv->base.commandLine = argv->argv;
+			memset(argv->argv, 0, sizeof(argv->argv));
+			bool isSd = device->phisicalDrive == 0;
```

Comment:
the `0` can be replaced with the existing `SDMMC_DEVICE_SD` define in `arm9/sdmmc.h`, though it would have to be replicated in `arm7/sdmmc.h`

Reply:
The `0` here comes from the device list "specification", from gbatek `0    Physical Drive (0=External SD/MMC Slot, 1=Internal eMMC)`, technically it's not mapped to the values used in libnds

Reply:
Then you could add additional defines or enums like `DEVICELIST_DEVICE_SD` and `DEVICELIST_DEVICE_NAND`. This would be more readable